aten/src/ATen/native/cuda/layer_norm_kernel.cu

```diff
@@ -852,21 +852,21 @@ std::tuple<Tensor, Tensor, Tensor> layer_norm_cuda(
   Tensor rstd = at::empty({M}, X->options().dtype(acc_type));
   if (M > 0) {
     LayerNormKernelImpl(*X, *gamma, *beta, M, N, eps, &Y, &mean, &rstd);
+  }
+  const auto input_shape = input.sizes();
```

Comment:
Let's add a comment for this block -- is this a BC-breaking change?

Reply:
> is this a BC-breaking change?
Yes, it does change the shape of var and mean returned for input tensors with 0s in the shape. It's a bug-fix.

Reply:
Behavior on master:
```py
In [1]: import torch

In [2]: a = torch.randn(2, 0, 3, 3, device='cuda') # note 0 in the 2nd dim

In [3]: [x.shape for x in torch.native_layer_norm(a, (3, 3), None, None, 1e-5)]
Out[3]: [torch.Size([2, 0, 3, 3]), torch.Size([0]), torch.Size([0])] # shapes for var and mean are wrong

In [4]: a = torch.randn(2, 1, 3, 3, device='cuda') # note non-zero in the 2nd dim

In [5]: [x.shape for x in torch.native_layer_norm(a, (3, 3), None, None, 1e-5)]
Out[5]: [torch.Size([2, 1, 3, 3]), torch.Size([2, 1, 1, 1]), torch.Size([2, 1, 1, 1])]
```